### PR TITLE
Create domains through the partner API

### PR DESCRIPTION
**Pull request by stefanor**
@RayeN: I think this does what we need to create zones through the partner API.

I can't figure out a sane way to test the model function (the fakes don't share state between each other, and the tests assume this)

Feel free to hack on it, merge it, and release it.

### CODE REVIEW

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: pycloudflare/models.py [hunk 1]
```diff
@@ -7,31 +8,40 @@
 
 class User(object):
     def __init__(self, email, api_key):
-        self.api_key = api_key
         self.email = email
-        self.service = self.get_service(email, api_key)
+        self.api_key = api_key
```
Comment (stefanor) [inline pycloudflare/models.py]: Unused?

File: pycloudflare/models.py [hunk 2]
```diff
@@ -7,31 +8,40 @@
...
+        return cls.create_from_api_response(data)
+
+    @classmethod
+    def create_from_api_response(cls, data):
+        user = User(data['cloudflare_email'], data['user_api_key'])
+        set_property_cache(user, 'data', data)
+        return user
+
+    @cached_property
+    def data(self):
```
Comment (stefanor) [inline pycloudflare/models.py]: partner_data?

File: pycloudflare/models.py (hunk 1 above)
Comment (RayeN) [inline pycloudflare/models.py]: Right, just thought it might be used by clients.  But it's probably not.  Will remove.

Comment (RayeN) [inline pycloudflare/models.py]: but... Actually, `api_key` was a part of interface previously.  Probably we should leave it for consistency?

Comment (stefanor) [inline pycloudflare/models.py]: Changing it won't affect anything, so I'd drop it. It's not a very useful part of the interface.

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by bearnard** [comment]
LGTM :+1:

**Pull request merged**